crawl-ref/source/dgn-overview.cc

```diff
@@ -1094,3 +1064,164 @@ bool connected_branch_can_exist(branch_type br)
 
     return true;
 }
+
+static int _prompt_annotate_branch(level_id lid)
+{
+    // XXX I'd be grateful if someone could refactor this into something that's
+    // not black magic. The functionality doesn't have the be the exact same,
+    // so long as the list is of reasonable length.
```

Comment:
which is the black magic bit? the branch_iterator stuff? or the prompt message printing?

Reply:
Apparently you understand what's happening.
I don't understand `branch_iterator`, `**it`, and `it->id`

Reply:
er, i think `(**it).id` and `it.id` are equivalent, so please change the former one to the latter.
`branch_iterator` is a helper class that runs through the list of all branches, `it` is an instance of this iterator that represents the current position in the list of all branches, and `it->id` gets you the branch id (`enum branch_type`) for that position.

Reply:
I tried compiling with it.id instead of (**it).id, and it threw an error: `‘class branch_iterator’ has no member named ‘id’`

Reply:
ah, sorry, meant to say `it->id` not `it.id`

Reply:
it works